src/engine/strat_engine/metadata.rs

```diff
@@ -74,14 +74,22 @@ impl BDA {
         Ok(())
     }
 
+    /// Save metadata to the disk
     pub fn save_state(&mut self,
                       time: &Timespec,
                       metadata: &[u8],
                       mut f: &mut File)
                       -> EngineResult<()> {
-        try!(self.regions.save(time, metadata, &mut f));
+        self.regions.save_state(time, metadata, &mut f)
+    }
 
-        Ok(())
+    /// Read latest metadata from the disk
+    pub fn load_state(&self, mut f: &mut File) -> EngineResult<Option<Vec<u8>>> {
+        self.regions.load_state(&mut f)
+    }
+
+    pub fn last_updated(&self) -> &Option<Timespec> {
```

Comment:
I think the name could be improved to something like ```last_update_time```. Otherwise, it sounds a bit like you're locating the last updated thing.